### DISCUSSION TITLE
[New Feature] Officers

### DISCUSSION THREAD

**Discussion by CCTiberius** [Vinifera-Developers/Vinifera#1509] [Feature Requests]
### Description:
Basically a key you can add to an infantry flagging them as an officer, making them boost the stats of infantry around them. Would be great for having a way to boost early infantry into being lategame-relevant, and would add some new tactics.

### Possible Implementation:
An INI key like `Officer=Yes` maybe along with a global like `OfficerAuraRadius=`

**Comment by Rampastring** [comment]
Instead of limiting this to infantry, I'd make it a `TechnoType` key so it can also be added to buildings, vehicles and even aircraft.

This would also need keys for customizing the officer effects (more damage / speed / armor / whatever), and keys for defining what object types the effect is applied to (infantry/vehicles/buildings/aircraft).

**Comment by Crimsonum** [comment]
^What Rampa said. This seems like a singular case of what could be a more generic, area-effect, status modifier system. Think of propaganda towers/trucks that boost nearby units, superweapons that apply temporary (de)buffs, etc.

**Comment by CCTiberius** [comment]
I was gonna get to AoE-debuffs and treat that as a separate thing entirely, but if it is easier for the crew to make a universal system like this then go for it ofc.

**Comment by Rampastring** [comment]
I'm not very familiar with Ares, but doesn't it implement status modifiers with AttachEffect? Maybe someone could check its implementation and whether it could be a good base for (de)buffs in TS as well.

This indeed could then work as a way of applying effects to all units in a large area. The same mechanic could also be re-used for simple weapons that apply (de)buffs.

**Comment by CCTiberius** [comment]
I am pretty sure I have heard of that yes, someone told me of a smoke screen applying a damage debuff to anyone within it made with Ares logic

**Comment by GraionDilach** [comment]
AttachEffect should be possible in Vinifera.

What I did back then was to create INI tags on `Warhead` and `TechnoType` which affects the crate-firepower/armor/cloak/speed variables and then hook into the crate-pickup code to move their changes into `Ext`. Auras in Ares are done with people using constantly-fired-and-immediately-exploding dummy weapons.

I'm not sure how Alex did the ROF effect.
